projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -97,13 +97,17 @@ export class IgxFilteringService implements OnDestroy {
                     filterCell.updateFilterCellArea();
                 });
             });
+        }
+    }
 
-            this.grid.onColumnVisibilityChanged.pipe(takeUntil(this.destroy$)).subscribe((eventArgs: IColumnVisibilityChangedEventArgs) => {
-                if (this.grid.filteringRow && this.grid.filteringRow.column === eventArgs.column ) {
-                    this.grid.filteringRow.close();
+    /**
+     * Close filtering row if a column is hidden.
+     */
+    public hideColumnFilteringRowOnVisibilityChange(col: IgxColumnComponent) {
+        const filteringRow = col.grid.filteringRow;
 
-                }
-            });
+        if (filteringRow && filteringRow.column) {
```

Comment:
Add a check that `filteringRow.column === col`. Otherwise we are going to close the filtering row on any column hiding.